HADOOP-16895. [thirdparty] Revisit LICENSEs and NOTICEs

vinayakumarb opened the pull request.

vinayakumarb commented:
@aajisaka Could you please review this?

aajisaka commented:
### Source release
We don't have to include the dependencies in LICENSE and NOTICE files.

### Binary release
LICENSE-binary and NOTICE-binary should contain protobuf, jaegartracing, and all their dependencies (such as gson). In addition, we should include LICENSE-binary and NOTICE-binary files in the jar files.

Details: http://www.apache.org/dev/licensing-howto.html

aajisaka commented:
FYI: https://github.com/apache/hbase-thirdparty/commit/0a3913706982e5355bf2331dc6933a18a0bb86e8

vinayakumarb commented:
Thanks @aajisaka, I will update accordingly.

vinayakumarb commented:
@aajisaka  Updated the PR to include License-binary and NOTICE-binary as LICENCE.txt and NOTICE.txt and all other Non-ALV2 licenses under /META-INF/licences-binary inside shaded jars.

Please review.

vinayakumarb commented:
Thanks @elek 
I have made NOTICE-binary 'lightweight' by removing 'non-mandatory' mentions of all 3rd-party dependencies.
* Apache Thrift and Kotlin do provide NOTICE files. So copied kept those contents alone in NOTICE-binary.
* jetbrains' java-annotation alone have copyright mention in their ALv2 txt, so copied entire licence under licences-binary.
* keeping both NOTICE.txt and NOTICE-binary in jars.

Hope I have addressed all your concerns. Please review.

vinayakumarb commented:
Thanks @elek and @aajisaka for reviews. Merged this to trunk and will cherry-pick to branch-1.0 for another RC

elek commented:
Thank you very much the work @vinayakumarb (And sorry for the occasionally delayed answer...)
